File: Data Structures & Algorithms/valid-binary-search-tree/submission-2.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Value {
    int minValue = Integer.MAX_VALUE;
    int maxValue = Integer.MIN_VALUE;
} 

class Solution {
    public boolean isValidBST(TreeNode root) {
       return isValidBSTH(root, Long.MIN_VALUE, Long.MAX_VALUE);
    }


    public boolean isValidBSTH(TreeNode root, long left, long right) {
        
        if(root == null) return true;

        if (!(left < root.val && root.val < right)) {
            return false;
        }

        return isValidBSTH(root.left, left, root.val) && isValidBSTH(root.right,root.val, right);
         
    }
}

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        
        if not root:
           return []

        res_final: List[List[int]] = []

        queue: List[Optional[TreeNode]] = []

        queue.append(root)
        queue.append(None)

        res = []
        while queue:
            node = queue.pop(0)
            if node:
                res.append(node.val)
                if node.left:
                    queue.append(node.left)
                if node.right:
                    queue.append(node.right)    
            else:
                res_final.append(res)
                res = []

                if queue:
                    queue.append(None)
            
        return res_final

        